content/source/docs/enterprise/sentinel/import/mock-tfstate.html.md

```diff
@@ -22,127 +22,195 @@ documentation][ref-official-sentinel-documentation].
-          "value": "bar"
-        },
-        "list": {
-          "sensitive": false,
-          "type": "list",
-          "value": [
-            "foo",
-            "bar"
-          ]
+    },
+    "map": {
+      "sensitive": false,
+      "type":      "map",
+      "value": {
+        "foo":    "bar",
+        "number": 42,
+      },
+    },
+    "string": {
+      "sensitive": false,
+      "type":      "string",
+      "value":     "foo",
+    },
+  },
+  "path": [],
+  "resources": {
+    "null_resource": {
+      "bar": {
+        0: {
+          "attr": {
+            "id": "5511801804920420639",
+            "triggers": {
+              "foo_id": "3332481276939020374",
+            },
+          },
+          "depends_on": [
+            "null_resource.foo",
+          ],
+          "id":      "5511801804920420639",
+          "tainted": false,
         },
-        "map": {
-          "sensitive": false,
-          "type": "map",
-          "value": {
-            "foo": "bar",
-            "number": 42
-          }
+      },
+      "foo": {
+        0: {
+          "attr": {
+            "id": "3332481276939020374",
+            "triggers": {
+              "foo": "bar",
+            },
+          },
+          "depends_on": [],
+          "id":         "3332481276939020374",
+          "tainted":    false,
         },
-        "string": {
-          "sensitive": false,
-          "type": "string",
-          "value": "foo"
-        }
       },
-      "resources": {
-        "null_resource": {
-          "bar": {
-            "0": {
-              "depends_on": [
-                "null_resource.foo"
-              ],
-              "id": "5511801804920420639",
-              "attr": {
-                "id": "5511801804920420639",
-                "triggers": {
-                  "foo_id": "3332481276939020374"
-                }
-              },
-              "tainted": false
-            }
+    },
+  },
+}
+
+module_foo = {
+  "data":    {},
+  "outputs": {},
+  "path": [
+    "foo",
+  ],
+  "resources": {
+    "null_resource": {
+      "foo": {
+        0: {
+          "attr": {
+            "id": "7102772838417697789",
+            "triggers": {
+              "foo": "bar",
+            },
           },
-          "foo": {
-            "0": {
-              "depends_on": null,
-              "id": "3332481276939020374",
-              "attr": {
-                "id": "3332481276939020374",
-                "triggers": {
-                  "foo": "bar"
-                }
-              },
-              "tainted": false
-            }
-          }
-        }
-      }
-    }
+          "depends_on": [],
+          "id":         "7102772838417697789",
+          "tainted":    false,
+        },
+      },
+    },
+  },
+}
+
+module_paths = [
+  [],
+  [
+    "foo",
+  ],
+]
+
+terraform_version = "0.11.7"
+
+module = func(path) {
+  if length(path) == 0 {
+    return _root
+  }
+  if length(path) == 1 and path[0] is "foo" {
```

Comment:
This seems REAL squirrelly. As a user, I want something more resilient that I can copy-paste... can we give people a general purpose `module()` function for mocks? Like, can we save the module data as a map whose keys are module paths, and just index into that?

Reply:
These mocks are auto-generated so readability might not be 100% optimal.

The function could probably be optimized in the way that you are proposing, but I still would like it pushed for now so that we have at least communicated that mocking all parts of the Terraform imports are now possible. These can be updated at a later time.